actor-server/actor-bots/src/main/scala/im/actor/server/bot/http/BotsHttpHandler.scala

```diff
@@ -1,100 +1,109 @@
-package im.actor.server.api.http.bots
+package im.actor.server.bot.http
 
 import java.util.Base64
 
 import akka.actor.ActorSystem
 import akka.http.scaladsl.model.StatusCodes._
-import akka.http.scaladsl.model.ws.{ BinaryMessage, TextMessage, Message }
+import akka.http.scaladsl.model.ws.{ BinaryMessage, Message, TextMessage }
 import akka.http.scaladsl.model.{ HttpMethod, StatusCode }
 import akka.http.scaladsl.server.Directives._
 import akka.http.scaladsl.server.Route
-import akka.stream.Materializer
+import akka.stream.{ ActorMaterializer, Materializer }
 import akka.stream.scaladsl.Flow
 import akka.util.ByteString
 import cats.data.OptionT
 import cats.std.future._
 import de.heikoseeberger.akkahttpplayjson.PlayJsonSupport
 import im.actor.api.rpc.sequence.UpdateRawUpdate
-import im.actor.server.api.http.RoutesHandler
+import im.actor.server.api.http.HttpHandler
 import im.actor.server.api.http.json.{ ContentUnmarshaller, JsValueUnmarshaller, JsonFormatters, Status }
-import im.actor.server.bot.{ BotExtension, BotServerBlueprint }
+import im.actor.server.bot.{ BotServerBlueprint, BotExtension }
 import im.actor.server.model.AuthSession
 import im.actor.server.user.UserExtension
-import play.api.libs.json._
+import play.api.libs.json.{ JsNull, JsObject, JsString }
 import upickle.default._
 
 import scala.concurrent.Future
 import scala.util.control.NoStackTrace
 import scala.util.{ Failure, Success }
 
-private[http] final class BotsHandler(implicit system: ActorSystem, val materializer: Materializer) extends RoutesHandler with PlayJsonSupport with JsValueUnmarshaller with ContentUnmarshaller {
+private[bot] final class BotsHttpHandler(implicit system: ActorSystem)
+  extends HttpHandler
+  with PlayJsonSupport
+  with JsValueUnmarshaller
+  with ContentUnmarshaller {
 
   import JsonFormatters._
   import im.actor.bots.BotMessages.{ BotRequest, BotResponse, BotUpdate }
   import system._
 
-  private val botExt = BotExtension(system)
+  implicit val materializer: Materializer = ActorMaterializer()
+
+  //todo: dangerous dependency. move it somehow
+  private lazy val botExt = BotExtension(system)
```

Comment:
Couldn't find better way. The danger is: we use `BotExtension` in this httpHandler, and this httpHandler is registered on `BotExtension` initialization